ArchiSteamFarm/Core/ASF.cs

```diff
@@ -779,36 +781,37 @@ public static class ASF {
 		await UpdateSemaphore.WaitAsync().ConfigureAwait(false);
 
 		try {
-			// If backup directory from previous update exists, it's a good idea to purge it now
-			string backupDirectory = Path.Combine(SharedInfo.HomeDirectory, SharedInfo.UpdateDirectory);
+			try {
+				// If directories from previous update exist, it's a good idea to purge them now
+				foreach (string directory in Directory.EnumerateDirectories(SharedInfo.HomeDirectory, $"{SharedInfo.UpdateDirectoryNewPrefix}*")) {
+					ArchiLogger.LogGenericInfo(Strings.UpdateCleanup);
 
-			if (Directory.Exists(backupDirectory)) {
-				ArchiLogger.LogGenericInfo(Strings.UpdateCleanup);
+					Directory.Delete(directory, true);
 
-				for (byte i = 0; (i < WebBrowser.MaxTries) && Directory.Exists(backupDirectory); i++) {
-					if (i > 0) {
-						// It's entirely possible that old process is still running, wait a short moment for eventual cleanup
-						await Task.Delay(5000).ConfigureAwait(false);
-					}
+					ArchiLogger.LogGenericInfo(Strings.Done);
+				}
 
-					try {
-						Directory.Delete(backupDirectory, true);
-					} catch (Exception e) {
-						ArchiLogger.LogGenericDebuggingException(e);
+				foreach (string directory in Directory.EnumerateDirectories(SharedInfo.HomeDirectory, $"{SharedInfo.UpdateDirectoryOldPrefix}*")) {
+					ArchiLogger.LogGenericInfo(Strings.UpdateCleanup);
 
-						continue;
-					}
+					await Utilities.DeletePotentiallyUsedDirectory(directory).ConfigureAwait(false);
 
-					break;
+					ArchiLogger.LogGenericInfo(Strings.Done);
 				}
 
+				string backupDirectory = Path.Combine(SharedInfo.HomeDirectory, SharedInfo.UpdateDirectoryOld);
+
 				if (Directory.Exists(backupDirectory)) {
-					ArchiLogger.LogGenericError(Strings.WarningFailed);
+					ArchiLogger.LogGenericInfo(Strings.UpdateCleanup);
+
+					await Utilities.DeletePotentiallyUsedDirectory(backupDirectory).ConfigureAwait(false);
 
-					return (false, null);
+					ArchiLogger.LogGenericInfo(Strings.Done);
 				}
```

Comment:
The same code already exists in `Utilities.UpdateFromArchive` and `PluginsCore.UpdatePlugin`, so I would suggest moving it to a separate function, something like `Utilities.UpdateCleanup`. Also user might be confused why `UpdateCleanup`/`Done` is printed multiple times, so probably we can check at first if we have anything to delete at all and print it just a single time.